plugins/destination/duckdb/client/migrate.go

```diff
@@ -0,0 +1,271 @@
+package client
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/cloudquery/plugin-sdk/schema"
+	"github.com/cloudquery/plugin-sdk/specs"
+)
+
+const (
+	sqlTableInfo = "PRAGMA table_info('%s');"
+)
+
+type columnInfo struct {
+	index        int
+	name         string
+	typ          string
+	notNull      bool
+	defaultValue any
+	pk           bool
+}
+
+type tableInfo struct {
+	columns []columnInfo
+}
+
+func (c *Client) duckdbTables(tables schema.Tables) (schema.Tables, error) {
+	allTables := tables.FlattenTables()
+	var schemaTables schema.Tables
+	for _, table := range allTables {
+		info, err := c.getTableInfo(table.Name)
+		if err != nil {
+			return nil, err
+		}
+		if info == nil {
+			continue
+		}
+		schemaTable := schema.Table{
+			Name: table.Name,
+		}
+		for _, col := range info.columns {
+			schemaTable.Columns = append(schemaTable.Columns, schema.Column{
+				Name: col.name,
+				Type: c.duckdbTypeToSchema(col.typ),
+				CreationOptions: schema.ColumnCreationOptions{
+					PrimaryKey: col.pk,
+					NotNull:    col.notNull,
+				},
+			})
+		}
+		schemaTables = append(schemaTables, &schemaTable)
+	}
+
+	return schemaTables, nil
+}
+
+func (c *Client) normalizeColumns(tables schema.Tables) schema.Tables {
+	allTables := tables.FlattenTables()
+	var normalized schema.Tables
+	for _, table := range allTables {
+		tableCopy := table.Copy(table.Parent)
+		for i := range tableCopy.Columns {
+			// In DuckDB, a PK column must be NOT NULL, so we need to make sure that the schema we're comparing to has the same
+			// constraint.
+			if !c.enabledPks() {
+				tableCopy.Columns[i].CreationOptions.PrimaryKey = false
+			} else if tableCopy.Columns[i].CreationOptions.PrimaryKey {
+				tableCopy.Columns[i].CreationOptions.NotNull = true
+			}
+			// Since multiple schema types can map to the same duckdb type we need to normalize them to avoid false positives when detecting schema changes
+			tableCopy.Columns[i].Type = c.duckdbTypeToSchema(c.SchemaTypeToDuckDB(table.Columns[i].Type))
+		}
```

Comment:
```suggestion
		for i, col := range tableCopy.Columns {
			// In DuckDB, a PK column must be NOT NULL, so we need to make sure that the schema we're comparing to has the same
			// constraint.
			switch {
				case !c.enabledPks():
					tableCopy.Columns[i].CreationOptions.PrimaryKey = false
				case col.CreationOptions.PrimaryKey:
					tableCopy.Columns[i].CreationOptions.NotNull = true
			}
			// Since multiple schema types can map to the same duckdb type we need to normalize them to avoid false positives when detecting schema changes
			tableCopy.Columns[i].Type = c.duckdbTypeToSchema(c.SchemaTypeToDuckDB(col.Type))
		}
```